plugins/docker_binary_builder/README.md

```diff
@@ -0,0 +1,63 @@
+# Docker Builder for Binaries
+
+This plugin is useful when your projects are based on compiled languages and you don't want to base
+your deployment docker images on images that have all the tools/libs/compilers installed.
+
+Advantages to separating your docker images include:
+- Smaller deployment docker images.
+- Better security in your docker images to not be able to recompile code.
+
+## Getting Started
+
+This plugin makes a few assumptions:
+- You have 2 dockerfiles:
+ - 'Dockerfile' used to generate your main deployment docker image
+ - 'Dockerfile.build' used to create a temporary image to build any binaries you want to package into the former.
+- You have a build script 'build.sh' that can be executed to compile the needed binaries
+  - This is packaged inside the Dockerfile.build image and executed
```

Comment:
In particular, it looks like it's required that the `Dockerfile.build` creates an image with a file called `/app/build.sh`, because you have that hard-coded as the CMD when running the container.

Rather than that, how about you expect that the image created by `Dockerfile.build` specifies its own `CMD` instruction, and then when you run it as a container, you don't override the `CMD`?  That would make for one less file that needs to be named something special, and gives some more flexibility to the person writing the build image.